docs/cost-optimisation.md

```diff
@@ -0,0 +1,148 @@
+  podGC:
+    strategy: OnPodCompletion
+```
+
+You can set these configurations globally using [Default Workflow Spec](default-workflow-specs.md).
+
+If you need to keep records historically, use the [Workflow Archive](workflow-archive.md).
+
+Changing these settings will not delete workflows that have already run. To list old workflows:
+
+```
+argo list --completed --since 7d
+```
+
+> v2.9 and after
+
+To list/delete workflows completed over 7 days ago:
+
+```
+argo list --older 7d
+argo delete --older 7d
+```
+
+## Set Resources Requests Of Your Argo Instances
+
+> Suitable if you have many instances, e.g. on dozens of clusters or namespaces.
+
+Set a resource quota for the namespace you install Argo in to limit its total usage, e.g.
+
+```
+apiVersion: v1
+kind: ResourceQuota
+metadata:
+  name: argo
+spec:
+  hard:
+    pods: "4"
+    limits.cpu: 1000m
+    limits.memory: 1Gi
+    requests.cpu: 500m
+    requests.memory: 128Mi
+```
+
+Use limit range to set default container requests and limits, e.g.
+
+```
+apiVersion: v1
+kind: LimitRange
+metadata:
+  name: argo
+spec:
+  limits:
+    - type: Container
+      defaultRequest:
+        cpu: 100m
+        memory: 64Mi
+      default:
+        cpu: 500m
+        memory: 128Mi
+```
+
+This above limit is suitable for the Argo Server, as this is stateless. The Workflow Controller is stateful and will scale to the number of live workflows - so you are likely to need higher values.
+
+## Configure Executor Resource Requests
+
+> Suitable for all - unless you have large artifacts.
+
+Configure [workflow-controller-configmap.yaml](workflow-controller-configmap.yaml) to set the `executorResources`
+
+```
+    executorResources:
+      requests:
+        cpu: 100m
+        memory: 64Mi
+      limits:
+        cpu: 500m
+        memory: 512Mi
+```
+
+The correct values depend on the size of artifacts your workflows download. For artifacts >10GB, memory usage maybe large - [#1322](https://github.com/argoproj/argo/issues/1322).
+
+## Set The Workflows Pod Resource Requests 
+
+> Suitable if you are running a workflow with many homogenous pods.
+
+[Resource duration](resource-duration.md) shows the amount of CPU and memory requested by a pod and is indicative of the cost. You can use this to find costly steps within your workflow.
+
+Smaller requests can be set in the pod spec patch's [resource requirements](fields.md#resourcerequirements). 
+
+## Use A Node Selector To Use Cheaper Instances
+
+You can use a [node selector](fields.md#nodeselector) for cheaper instances, e.g. spot instances:
+
+```
+nodeSelector:
+  "node-role.kubernetes.io/argo-spot-worker": "true"
+```
+
+## Use Volume Claims Templates Instead Of Artifacts
+
+> Suitable if you have a workflow that passes a lot of artifacts within itself.
+
+Copying artifacts to and from storage outside of a cluster can be expensive. You can use a volume claim template to mount a volume that is attached to each step within the cluster
```

Comment:
> Suitable if you have a workflow that passes a lot of artifacts within itself.

Theres so many variables in this that it's hard to make any recommendation other than "consider doing this depending on this."

Volumes vs. VolumeClaimTemplates vs. Artifacts is highly dependent on what their artifact storage provider is vs. what volume they are using. For example, I believe it is more expensive to allocate and delete new EBS volumes every workflow using the PVC feature, than it is to upload and download some small files to S3.

On the other hand if they are using a NFS volume shared between all their workflows with large artifacts, that might be cheaper than the data transfer and storage costs of S3.

Lets reword this so we outline the variables to consider when choosing when to use artifacts vs. volumes:
1. data transfer costs (upload/download vs. copying)
2. data storage costs (s3 vs. volume)
3. requirement for parallel access to data (NFS vs. EBS vs. artifact)